.github/workflows/main.yml

```diff
@@ -1,6 +1,10 @@
 name: Build & Test
 
-on: [push]
+on:
+  push:
+    branches: main
+  pull_request:
+    branches: '**'
 
 jobs:
   test:
```

Comment:
Actually, I've now noticed that this workflow depends on some secrets and those are not available to pull requests created by outside collaborators.

I think that Turbo should be able to fall back to just executing everything without using its remote cache (that's acceptable to me) and Codecov doesn't require the token for public repos... but you still provide it:
https://github.com/trpc/trpc/blob/22f268d1a16daa9154db2a456323366471fe30d7/.github/workflows/main.yml#L57

Can that explicit token just be removed?